src/hotspot/share/runtime/arguments.cpp

```diff
@@ -1055,13 +1055,15 @@ void Arguments::add_string(char*** bldarray, int* count, const char* arg) {
     return;
   }
 
-  int new_count = *count + 1;
```

Comment:
Reviewer hint: I changed this since the original realloc had been inefficient and causing way too many reallocations.